fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -293,11 +295,26 @@ private void loadFromProto(@Nonnull RecordMetaDataProto.MetaData metaDataProto,
                                @Nonnull Descriptors.FileDescriptor[] dependencies,
                                boolean processExtensionOptions) {
         recordsDescriptor = buildFileDescriptor(metaDataProto.getRecords(), dependencies);
+        loadSubspaceKeySettingsFromProto(metaDataProto);
         unionDescriptor = initRecordTypes(recordsDescriptor, processExtensionOptions);
         loadProtoExceptRecords(metaDataProto);
         processSchemaOptions(processExtensionOptions);
     }
 
+    private void loadSubspaceKeySettingsFromProto(RecordMetaDataProto.MetaData metaDataProto) {
+        if (metaDataProto.hasSubspaceKeyCounter() && !metaDataProto.hasUsesSubspaceKeyCounter()) {
+            throw new MetaDataProtoDeserializationException(new MetaDataException("subspaceKeyCounter is set but usesSubspaceKeysCounter is not set in the meta-data proto"));
+        }
+        if (metaDataProto.hasUsesSubspaceKeyCounter() && !metaDataProto.hasSubspaceKeyCounter()) {
+            throw new MetaDataProtoDeserializationException(new MetaDataException("usesSubspaceKeysCounter is set but subspaceKeyCounter is not set in the meta-data proto"));
+        }
```

Comment:
I'm still not convinced this does the right thing if `subspace_key_counter` in the meta-data proto is set to false (as opposed to being unset). I think it would be sufficient to change the references to `hasUsesSubspaceKeyCounter` to just `getUsesSubspaceKeyCounter` if we are willing to rely on the fact that `get` on a boolean proto field returns false regardless of whether the field is unset or set explicitly to false.

Reply:
sure.